File: backoffice/src/product/domain/valueObjects/ProductStatus.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain import IntValueObject

"""
 *
 * Class 
 *
"""

class ProductStatus( IntValueObject ):

    """
     *
     * Constants 
     *
    """

    __ENABLED  = 1
    __DELETED  = 2

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : int ) -> None:
        super().__init__( value )
    
    @classmethod
    def createEnabled( cls ) -> object:
        return cls( cls.__ENABLED )
    
    @classmethod
    def createDeleted( cls ) -> object:
        return cls( cls.__DELETED )

File: preparation/src/product/domain/valueObjects/ProductStatus.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.product.domain {

    public class ProductStatus : IntValueObject {

        /* 
         * 
         * Constants
         *
        */

        private const int ENABLED = 1;
        private const int DELETED = 2;

        /* 
         * 
         * Methods
         *
        */

        public ProductStatus( int value ) : base( value ) {
        }

        public static ProductStatus createEnabled() {
            return new ProductStatus( ENABLED );
        }

        public static ProductStatus createDeleted() {
            return new ProductStatus( DELETED );
        }

    }

}